### ISSUE TITLE
AutomaticMaskGenerator

### ISSUE THREAD

**Issue by mavanish** [facebookresearch/sam3#226]
Is there utomaticMaskGenerator with Sam3. No need for text prompt!

**Comment by mario-koddenbrock** [comment]
Hey, yeah, you can totally do automatic mask generation with SAM3 without any prompts. The Hugging Face `transformers` library makes this pretty easy with its `mask-generation` pipeline, which basically finds and segments everything it can in an image.

Here's a quick example of how you'd call the pipeline directly:

```python
import numpy as np
from PIL import Image
from transformers import pipeline
import matplotlib.pyplot as plt
import torch

# A common issue is having an old version of transformers.
# Installing directly from the main branch on GitHub usually fixes things.
# pip install git+https://github.com/huggingface/transformers.git

# 1. Set up the pipeline for automatic mask generation
device = "cuda" if torch.cuda.is_available() else "cpu"
generator = pipeline("mask-generation", model="facebook/sam3", device=device)

# 2. Load your image
# Replace this with your actual image path, e.g., Image.open('path/to/image.png')
image = Image.fromarray(np.random.randint(0, 255, (256, 256, 3), dtype=np.uint8))

# 3. Run the generator on the image
# You can also pass other parameters like `points_per_batch` here
results = generator(image)
masks = results["masks"]

print(f"Found {len(masks)} masks in the image.")

# --- Optional: Helper function to see the results ---
def show_masks_on_image(img, mask_list):
    if not mask_list:
        print("No masks to show.")
        return
    plt.figure(figsize=(10, 10))
    plt.imshow(img)
    ax = plt.gca()
    ax.set_autoscale_on(False)

    for mask in mask_list:
        # Give each mask a random color
        color = np.concatenate([np.random.random(3), np.array([0.6])], axis=0)
        h, w = mask.shape[-2:]
        mask_image = mask.reshape(h, w, 1) * color.reshape(1, 1, -1)
        ax.imshow(mask_image)

    plt.axis('off')
    plt.savefig("sam3_automatic_masks.png")
    plt.show()

show_masks_on_image(image, masks)
```

A couple of extra things to keep in mind:

*   **It's a drop-in replacement for SAM2:** The pipeline is exactly the same. If you were using the automatic generator with SAM2 before, you can just swap the model name to `"facebook/sam3"` and you're good to go.
*   **Prompts are more powerful:** Honestly, while the automatic mode is cool, the real strength of SAM is using prompts (text, points, or boxes). You get way more control and usually better results. I'd highly recommend using prompts for any serious work.
*   **Installation issues:** If you run into errors, it's often because your `transformers` library is out of date. SAM3 is new, so you might need to grab the latest version directly from their GitHub:
    ```bash
    pip install git+https://github.com/huggingface/transformers.git
    ```

Hope this helps
